### DISCUSSION TITLE
How to change the depth and gain in fcal

### DISCUSSION THREAD

**Discussion by saigidde** [PlusToolkit/PlusLib#929] [Q&A]
![Screenshot 2022-04-04 154654](https://user-images.githubusercontent.com/102110796/161837440-94c02f68-e6f8-4082-9a92-ab16d2f0cb0a.png)
How to change the gain and depth in fcal while performing the spatial calibration?
I tried to change the settings in the ultrasound machine, it works well there, but when I come back to the fcal for spatial calibration, it shows the maximum depth rather than small depth. Not sure if I have to change anything?
See the attached image with depth being large

**Comment by danilogr** [comment]
My hunch here is that fcal works with a fixed depth. So if you change the
depth on the machine, you either have to perform another probe
image-tracker calibration or load a previous calibration for the selected
depth.

**Reply by lassoan** [reply]
That's correct, fCal calibrates at a single depth. fCal sets the depth based on the value that is specified in the configuration file - if the ultrasound system supports dynamic depth change from the external interface. Calibration at each depth is necessary because when you change the depth then usually the transducer position in the image changes, so you cannot just multiply the spacing but you also need to change the translation component of the calibration matrix.

Ultrasonix (and maybe Telemed and ClariUS) supports automatic calibration update when depth is changed, either via Plus or on the device, because there is API to get the transducer position in the image.
